src/Solver/Dynamic/Ida.cpp

```diff
@@ -110,48 +116,113 @@ namespace AnalysisManager
       return this->configureLinearSolver();
     }
 
+    /**
+     * @brief Configure the linear solver
+     *
+     * @note This currently uses pre-processor directives to set dense or sparse
+     * linear solvers
+     *
+     * @tparam ScalarT
+     * @tparam IdxT
+     */
     template <class ScalarT, typename IdxT>
     int Ida<ScalarT, IdxT>::configureLinearSolver()
     {
       int retval = 0;
+
+/// Todo - Implement a cleaner way to handle the Sparse versus Dense solvers
+#ifdef GRIDKIT_ENABLE_SUNDIALS_SPARSE
+      if (model_->hasJacobian())
+      {
+        this->configureLinearSolverSparse();
+      }
+      else
+      {
+        this->configureLinearSolverDense();
+      }
+#else
       if (model_->hasJacobian())
       {
-        sunindextype n   = static_cast<sunindextype>(model_->size());
-        sunindextype nnz = static_cast<sunindextype>((model_->getJacobian()).nnz());
+        /// Todo - Improve error handling capabilities and hasJacobian_ ownership
+        throw std::runtime_error("SUNDIALS is not configured with KLU, but the model has a (sparse) Jacobian.");
```

Comment:
Instead of throwing an error, could we just fall back on the dense linear solver?